### PR TITLE
Compile and bundle server-side code with webpack

### PR DESCRIPTION
**Pull request by lightnet328**
# Changes

- Compile and bundle server-side code with webpack
- Use `ts-node` only in development

### CODE REVIEW

File: webpack/server/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const ROOT = path.join(__dirname, '/../../')
+
+const config = {
+  mode: process.env.NODE_ENV,
+  target: 'node',
+  node: {
+    __dirname: true,
+    __filename: true,
+  },
+  externals: [nodeExternals()],
```
Comment (otofune) [inline webpack/server/webpack.config.js]: 👍

File: webpack/server/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  entry: path.join(ROOT, 'lib/app.ts'),
+  output: {
+    path: path.join(ROOT, 'dist/server'),
```
Comment (otofune) [inline webpack/server/webpack.config.js]: [Just wondering] Do you plan to compile client-side assets to dist/client? I wondered about why using more deep hierarchy.

File: webpack/server/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    filename: 'app.js',
+  },
+  resolve: {
+    plugins: [new TsconfigPathsPlugin({ configFile: path.join(ROOT, 'tsconfig.server.json') })],
+    modules: ['./node_modules'],
+    extensions: ['.ts', '.jsx', '.json'],
```
[the remark this answers is not in the document]
Comment (lightnet328) [reply webpack/server/webpack.config.js]: I've change it to `.js`.

File: webpack/server/webpack.config.js (hunk 2 above)
Comment (lightnet328) [reply webpack/server/webpack.config.js]: I don't have such a plan at the moment.
However, I just meant to show the domain explicitly.

File: nodemon.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "execMap": {
+    "ts": "ts-node --files -r tsconfig-paths/register --project tsconfig.server.json"
+  },
+  "watch": ["lib/", "locales/"],
```
Comment (sotarok) [inline nodemon.json]: And `common/` ?

File: Dockerfile [hunk 5]
```diff
@@ -32,4 +32,4 @@ WORKDIR /crowi
 
 COPY --from=builder --chown=node:node /crowi /crowi
 
-CMD ["node", "."]
+CMD ["npm", "run", "start"]
```
Comment (sotarok) [inline Dockerfile]: 👍

File: Procfile [hunk 6]
```diff
@@ -1 +1 @@
-web: node .
+web: npm run start
```
Comment (sotarok) [inline Procfile]: 👍  (npm start is also okay)

**Pull request merged**